### PR TITLE
Fuzzing: Move over two fuzzers from cncf-fuzzing

### PR DESCRIPTION
**Pull request by AdamKorcz**
Move over [these two token fuzzers](https://github.com/cncf/cncf-fuzzing/blob/main/projects/distribution/token_fuzzer.go) from cncf-fuzzing. The dependencies come from `github.com/AdaLogics/go-fuzz-headers` which is used to create random structs.

Signed-off-by: AdamKorcz <adam@adalogics.com>

### CODE REVIEW

**Review by corhere — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -74,3 +74,7 @@ require (
 	google.golang.org/grpc v1.31.0 // indirect
 	google.golang.org/protobuf v1.26.0 // indirect
 )
+
+require github.com/AdaLogics/go-fuzz-headers v0.0.0-20221103172237-443f56ff4ba8
```
Comment (thaJeztah) [inline go.mod:78]: Can you move this one into the first (direct dependencies) block and rerun go mod tidy? Go doesn't always group them correct, and needs a bit of a nudge)

File: go.mod [hunk 2]
```diff
@@ -74,3 +74,7 @@ require (
...
+
+require github.com/cyphar/filepath-securejoin v0.2.3 // indirect
```
Comment (thaJeztah) [inline go.mod:80]: same for this one, but this one is indirect so goes into the second block

**Review by thaJeztah — CHANGES_REQUESTED** [review]
thanks! left two comments; can you look at those (and amend / squash the commit)?

File: vendor/github.com/kr/pretty/License [hunk 3]
```diff
@@ -1,5 +1,3 @@
-The MIT License (MIT)
-
```
Comment (thaJeztah) [inline vendor/github.com/kr/pretty/License:2]: See https://github.com/kr/pretty/pull/62

**Review by thaJeztah — APPROVED** [review]
saw this one was still pending; I did a quick rebase, and fixed up the `go.mod` dependencies to be in the correct group

LGTM

**Comment by thaJeztah** [comment]
@milosgajdos @corhere as I was the last one pushing, perhaps one of you can (re-)LGTM and merge?

**Review by corhere — APPROVED** [review]

**Comment by milosgajdos** [comment]
@AdamKorcz can you rebase and run `go mod tidy` - I see some stuff in `go.sum` that I think should get cleaned up by `tidy`

**Comment by thaJeztah** [comment]
@milosgajdos I had the branch still checked out locally, but no diff on `go mod tidy`; anything specific that you were looking at?

**Comment by milosgajdos** [comment]
@thaJeztah is it normal to see this -- there are two versions of `kr/text` in the `go.sum` 🤔 

```
github.com/kr/text v0.1.0/go.mod h1:4Jbv+DJW3UT/LiOwJeYQe1efqtUx/iVham/4vfdArNI=
github.com/kr/text v0.2.0 h1:5Nx0Ya0ZqY2ygV366QzturHI13Jq95ApcVaJBhpS+AY=
github.com/kr/text v0.2.0/go.mod h1:eLer722TekiGuMkidMxC/pM04lWEeraHUUmBw8l2grE=
```

**Comment by thaJeztah** [comment]
Yeah, I think `go.sum` traverses all dependencies, then registers the checksum for each one it finds and uses to perform version resolution; https://go.dev/ref/mod#go-sum-files

> The go.sum file may contain hashes for multiple versions of a module. The go command may need to load go.mod files from multiple versions of a dependency in order to perform [minimal version selection](https://go.dev/ref/mod#minimal-version-selection). go.sum may also contain hashes for module versions that aren’t needed anymore (for example, after an upgrade). [go mod tidy](https://go.dev/ref/mod#go-mod-tidy) will add missing hashes and will remove unnecessary hashes from go.sum.

**Comment by milosgajdos** [comment]
Sadness. That's not how I thought it was working. Thanks for the link and the quote.

**Pull request merged**